model/textparse/interface.go

```diff
@@ -130,27 +167,33 @@ func extractMediaType(contentType, fallbackType string) (string, error) {
 // An error may also be returned if fallbackType had to be used or there was some
 // other error parsing the supplied Content-Type.
 // If the returned parser is nil then the scrape must fail.
-func New(b []byte, contentType, fallbackType string, parseClassicHistograms, convertClassicHistogramsToNHCB, skipOMCTSeries, enableTypeAndUnitLabels bool, st *labels.SymbolTable) (Parser, error) {
+func New(b []byte, contentType, fallbackType string, st *labels.SymbolTable, opts ...ParserOptions) (Parser, error) {
```

Comment:
b and content type are likely required arguments, but we could also pass `fallbackType` and especially `st *labels.SymbolTable` via new `WithFallbackType` and `WithLabelsSymbolTable` options. WDYT?